spec.bs

```diff
@@ -459,15 +462,29 @@ To <dfn>validate and convert auction ad config</dfn> given an {{AuctionAdConfig}
       Otherwise, [=list/append=] |buyer| to |buyers|.
   1. Set |auctionConfig|'s [=auction config/interest group buyers=] to |buyers|.
 1. If |config|["{{AuctionAdConfig/auctionSignals}}"] [=map/exists=]:
-  1. If |config|["{{AuctionAdConfig/auctionSignals}}"] is a {{Promise}}, let |auctionConfig|'s
-    [=auction config/auction signals=] be |config|["{{AuctionAdConfig/auctionSignals}}"].
+  1. If |config|["{{AuctionAdConfig/auctionSignals}}"] is a {{Promise}}:
+    1. Let |auctionConfig|'s [=auction config/auction signals=] be
+      |config|["{{AuctionAdConfig/auctionSignals}}"].
+    1. Increment |auctionConfig|'s [=auction config/pending promise count=].
+    1. [=Upon fulfillment=] of |auctionConfig|'s [=auction config/auction signals=] with |result|:
+      1. Set |auctionConfig|'s [=auction config/auction signals=] to the result of
+        [=serializing a JavaScript value to a JSON string=], given |result|.
+      1. Decrement |auctionConfig|'s [=auction config/pending promise count=].
   1. Otherwise, let |auctionConfig|'s [=auction config/auction signals=] be the result of
     [=serializing a JavaScript value to a JSON string=], given
     |config|["{{AuctionAdConfig/auctionSignals}}"].
-1. If |config|["{{AuctionAdConfig/sellerSignals}}"] [=map/exists=], let |auctionConfig|'s
-  [=auction config/seller signals=] of be the result of
-  [=serializing a JavaScript value to a JSON string=], given
-  |config|["{{AuctionAdConfig/sellerSignals}}"].
+1. If |config|["{{AuctionAdConfig/sellerSignals}}"] [=map/exists=]:
+  1. If |config|["{{AuctionAdConfig/sellerSignals}}"] is a {{Promise}}:
+    1. Let |auctionConfig|'s [=auction config/seller signals=] be
+      |config|["{{AuctionAdConfig/sellerSignals}}"].
+    1. Increment |auctionConfig|'s [=auction config/pending promise count=].
+    1. [=Upon fulfillment=] of |auctionConfig|'s [=auction config/seller signals=] with |result|:

```

Comment:
Out of curiosity what do we do in the implementation if the promise rejects? Do we reject the run ad auction promise too?

Reply:
We reject the runAdAuction() promise.

Reply:
OK then I think we have to do the same here in the spec, via "Upon rejection"